src/libraries/System.Net.Primitives/src/System/Net/Cookie.cs

```diff
@@ -307,6 +307,34 @@ private static bool IsDomainEqualToHost(string domain, string host)
                    (string.Compare(host, 0, domain, 1, host.Length, StringComparison.OrdinalIgnoreCase) == 0);
         }
 
+        internal static int CompareDomain(string? left, string? right)
+        {
+            if (left is null)
+            {
+                return right is null ? 0 : -1;
+            }
+
+            if (right is null)
+            {
+                return 1;
+            }
```

Comment:
Is `cookie.Domain == null` actually a valid case? To me it looks like the setter is always replacing `null` with `string.Empty`.

Reply:
Nice catch! I'll remove it